core/unit_test/TestMDRangePolicyConstructors.hpp

```diff
@@ -138,4 +138,50 @@ TEST(TEST_CATEGORY_DEATH, policy_invalid_bounds) {
 }
 #endif
 
+TEST(TEST_CATEGORY, policy_get_tile_size) {
+  constexpr int rank = 3;
+  using Policy    = Kokkos::MDRangePolicy<TEST_EXECSPACE, Kokkos::Rank<rank>>;
+  using tile_type = typename Policy::tile_type;
+
+  std::size_t last_rank =
+      (Policy::inner_direction == Kokkos::Iterate::Right) ? rank - 1 : 0;
+
+  auto default_size_properties =
+      Kokkos::Impl::get_tile_size_properties(TEST_EXECSPACE());
+
+  {
+    int dim_length = 100;
+    Policy policy({0, 0, 0}, {dim_length, dim_length, dim_length},
+                  tile_type{{2, 4, 16}});
+
+    auto rec_tile_sizes = policy.tile_size_recommended();
```

Comment:
Can we check that this matches `m_tile`?

Reply:
This shouldn't exact match `m_tile`. It will match for some ranks in this case (i.e. the first and last rank for host builds), but it depends on the backend.

The purpose of `tile_size_recommended` is to provide a hint on what MDRangePolicy would've used for tile sizes if tile sizes weren't passed in for the construction.
So, this unit-test is really just to check that `tile_size_recommended` still returns expected values even when constructed with tile sizes.

Reply:
OK. I meant can we check that we get the same tile sizes when using `team_size_recommended` and `m_tile` if the user didn't provide any tile sizes.

Reply:
Added a quick check to confirm that `m_tile` is consistent with what is returned by `team_size_recommended`.